### PR TITLE
[aes256] Typings for aes256

### PR DESCRIPTION
**Pull request by ThomasdenH**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/aes256/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    /**
+     * Decrypt an encrypted message back to clear-text using AES-256 plus a random Initialization Vector.
+     * @param encrypted  The encrypted message to be decrypted.
+     * @returns The original plain-text message.
+     */
+    decrypt(encrypted: string): string;
+}
+
+export default aes256;
```
Comment (plantain-00) [inline types/aes256/index.d.ts]: There is no `export default` https://github.com/JamesMGreene/node-aes256/blob/master/index.js#L163

**Review by plantain-00 — CHANGES_REQUESTED** [review]

File: types/aes256/index.d.ts (hunk 1 above)
Comment (ThomasdenH) [reply types/aes256/index.d.ts]: Changed it!

**Comment by ThomasdenH** [comment]
I can't figure out how to export both `const aes256` and the `namespace aes256 { class AesCipher }`. Maybe someone can help?

**Comment by plantain-00** [comment]
Can this link help? http://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-class-d-ts.html

**Comment by ThomasdenH** [comment]
The trouble is that while a class and a namespace can be declared simultaneously, a const and a namespace apparently can't. But declaring `aes256` as a class would be a mistake as well.

**Comment by ThomasdenH** [comment]
I have also read [https://github.com/Microsoft/TypeScript/issues/13536](this issue), and in particular the following example:

```typescript
declare class HTTPError { }

interface GotApi {
    (): any;
    delete(): any;
    HTTPError: typeof HTTPError;
}

declare var got: GotApi;
export = got;
```

However, I can't seem to reference `AesCipher` in any way. `aes256.AesCipher` gives the error that `aes256` is not a namespace.

**Pull request closed without merging**